src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Debugging/CSharpVirtualDocumentDebuggingExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Debugging
+{
+    internal static class CSharpVirtualDocumentDebuggingExtensions
+    {
+        // This method should only ever be used at debug language service time to ensure two super large assumptions:
```

Comment:
Also note: Once they've added an async debugging API we wouldn't technically need to worry about re-using C# syntax trees anymore. Also, once we're able to move to source generators we'll have direct access to syntax trees as well. This is all in an effort to mitigate a VS hang perf issue.